### ISSUE TITLE
How to Crop single image before calling torchvision.utils.save_image, If I am using PIL lib Image.crop(....) method then image quality degrade.

### ISSUE THREAD

**Issue by praveenkumarchandaliya** [pytorch/vision#1395]
vutils.save_image(fixed_fake.data,outputpath , normalize=True)

    print("output path",outputpath)
    img = Image.open(outputpath)
    noOfRow = 5
    noOfColumn = 8
    x1 = 2
    y1 = 2
    x2 = 130
    y2 = 130
    folder = file_batch

    for i in range(0, noOfColumn):
        dest_dir = file_batch[i].split("/")[7]
        if not os.path.exists(outf+"/"+dest_dir):
            os.mkdir(outf+"/"+dest_dir)
        for j in range(1, noOfRow + 1):
            area = (x1, y1, x2, y2)
            cropped_img = img.crop(area)
            imgName = "{}{}".format(i, j)
            cropped_img.save(os.path.join(outf+dest_dir,filename))
            y1 = y1 + 130
            y2 = y2 + 130
        x1 = x1 + 130
        x2 = x2 + 130
       y1 = 2
        y2 = 130

**Comment by pmeier** [comment]
What does degrade mean for you? Are you seeing compression artifacts for example from saving the image as `.jpg`? If that is the case try to save it in a format without compression:

```python
filename = os.path.splitext(filename)[0] + ".png"
cropped_img.save(os.path.join(outf+dest_dir, filename), compress_level=0)
```

If that does not solve your problem you need to provide more information for us to help you.

**Comment by praveenkumarchandaliya** [comment]
![ImageQualityQuestion](https://user-images.githubusercontent.com/27194129/66009308-f85be280-e487-11e9-850a-b018de1eb6b4.png)

**Comment by pmeier** [comment]
So you have a mini-batched tensor `BxCxHxW`, for example `B == 5` for 5 `images`, and you want to save each image individually? In that case I would simply [`split`](https://pytorch.org/docs/stable/torch.html#torch.split) the tensor instead the grid of images:

```python
import torch
from torchvision.utils import save_image

images = torch.rand(5, 3, 128, 128)

for num, image in enumerate(torch.split(images, 1)):
    save_image(image, "{}.jpg".format(num))
```

**Comment by SebastienEske** [comment]
Hello,
How about slicing? You need to do a bit of math but then you can be sure that there are no artifacts.
1. Crop of 128 with padding of 2 -> `step_size = 128+2`
2. Make sure that your image+2 (no need to pad the right of the rightmost crop) is a multiple of the step size:
```python
nbcropx = np.ceil((dimx+2)/step_size)
nbcropy = np.ceil((dimy+2)/step_size)
newdimx = nbcropx * step_size - 2
newdimy = nbcropy * step_size - 2
img = nn.functional.pad(img,(0, 0, newdimx-dimx, newdimy-dimy), mode='constant', value=0) #maybe x and y are reversed
```
3. And then you can just slice it:
```python
for i in range(nbcropx):
    for j in range(nbcropy):
        crop = img[..., i*step_size:i*step_size+128, j*step_size:j*step_size+128]
```
Simple and really no problem. I didn't test the code so there may be a few bugs but you get the idea. You can also improve it to add rescaling, etc...

**Comment by pmeier** [comment]
@SebastienEske If I understand @praveenkumarchandaliya correctly he just wants to save each image in a batch individually. He used `utils.save_image()`, which if given a batch creates a grid of images. Instead of creating a grid that you don't want and slicing / cropping / ... it afterwards, it is far easier to just save the images one by one.

**Comment by SebastienEske** [comment]
Oh! Ok. Sorry for my misunderstanding then.
